### PR TITLE
[WIP] [aarch64] Add CUDA 12.4 build workflow for ARM

### PR DESCRIPTION
**Pull request by tinglvv**
Redo from https://github.com/pytorch/builder/pull/1545/ to include only Cuda 12.4 build support for ARM wheel
WIP as it is dependent on https://github.com/pytorch/builder/pull/1722

### CODE REVIEW

**Comment by ptrblck** [comment]
@tinglvv the `lint` failure is real and it seems you are re-importing `shutil`.

File: common/install_cuda_aarch64.sh [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  wget -q https://developer.download.nvidia.com/compute/cudnn/redist/cudnn/linux-sbsa/cudnn-linux-sbsa-8.9.2.26_cuda12-archive.tar.xz -O cudnn-linux-x86_64-8.9.2.26_cuda12-archive.tar.xz
+  tar xf cudnn-linux-sbsa-8.9.2.26_cuda12-archive.tar.xz
+  cp -a cudnn-linux-sbsa-8.9.2.26_cuda12-archive/include/* /usr/local/cuda/include/
+  cp -a cudnn-linux-sbsa-8.9.2.26_cuda12-archive/lib/* /usr/local/cuda/lib64/
+  cd ..
+  rm -rf tmp_cudnn
+
+  # NCCL license: https://docs.nvidia.com/deeplearning/nccl/#licenses
+  # Follow build: https://github.com/NVIDIA/nccl/tree/v2.19.3-1?tab=readme-ov-file#build
+  git clone -b v2.19.3-1 --depth 1 https://github.com/NVIDIA/nccl.git
```
Comment (bryantbiggs) [inline common/install_cuda_aarch64.sh:26]: This was recently updated to 2.20.5 https://github.com/pytorch/builder/pull/1724
```suggestion
  git clone -b v2.20.5-1 --depth 1 https://github.com/NVIDIA/nccl.git
```

File: common/install_cuda_aarch64.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+set -ex
+
+function install_124 {
+  echo "Installing CUDA 12.4 and cuDNN 8.9 and NCCL 2.19.3 and cuSparseLt-0.5.2"
+  rm -rf /usr/local/cuda-12.4 /usr/local/cuda
+  # install CUDA 12.4.0 in the same container
+  wget -q https://developer.download.nvidia.com/compute/cuda/12.4.0/local_installers/cuda_12.4.0_550.54.14_linux_sbsa.run
```
Comment (bryantbiggs) [inline common/install_cuda_aarch64.sh:9]: should add a local variable to ensure the correct version is propagated throughout the function
```suggestion
  local cuda_version='12.4.0_550.54.14`
  wget -q "https://developer.download.nvidia.com/compute/cuda/12.4.0/local_installers/cuda_${cuda_version}_linux_sbsa.run"
```

Same for the other places below, and similarly for CUDNN

Comment (tinglvv) [reply common/install_cuda_aarch64.sh:9]: Thanks for the suggestion, that's a good point. Planning to refactor as the next step, but for this PR targeting to make the wheel work.

File: common/install_cuda_aarch64.sh (hunk 1 above)
Comment (tinglvv) [reply common/install_cuda_aarch64.sh:26]: Going to keep in sync with https://github.com/pytorch/builder/pull/1720/files

**Comment by tinglvv** [comment]
Closing as the commits are intertwined.

File: aarch64_linux/aarch64_wheel_ci_build.py [hunk 3]
```diff
@@ -9,100 +9,275 @@
...
-    '''
+    """
+    """
     wheel_name = list_dir(f"/{folder}/dist")[0]
 
     if "pytorch" in folder:
         print("Repairing Wheel with AuditWheel")
-        check_call(["auditwheel","repair", f"dist/{wheel_name}"], cwd=folder)
+        check_call(["auditwheel", "repair", f"dist/{wheel_name}"], cwd=folder)
+        check_call(["auditwheel", "repair", f"dist/{wheel_name}"], cwd=folder)
```
Comment (ptrblck) [inline aarch64_linux/aarch64_wheel_ci_build.py:106]: This seems to be duplicated

Comment (tinglvv) [reply aarch64_linux/aarch64_wheel_ci_build.py:106]: code is a bit messy in this PR due to rebase, will open a new PR with cleaned up code

**Pull request closed without merging**